pinc/post_processing.inc

```diff
@@ -0,0 +1,78 @@
+<?php
+include_once($relPath."project_states.inc"); // PROJ_POST_FIRST_CHECKED_OUT
+include_once($relPath."misc.inc"); // array_get
+
+// Global variable that specifies when we alert PPers about their
+// outstanding projects. Time is calculated as this many days prior
+// to midnight of the 1st of the current month.
+global $pp_alert_threshold_days;
+$pp_alert_threshold_days = 90;
+
+function count_pp_projects_past_threshold($PPer=NULL)
+{
+    $columns = [ "count(*) AS total" ];
+    $result = _run_pp_threshold_query_result($columns, $PPer);
+    
+    list($total) = mysqli_fetch_row($result);
+    return $total;
+}
+
+function get_pp_projects_past_threshold($PPer=NULL)
+{
+    $columns = [
+        "nameofwork",
+        "checkedoutby",
+        "projects.projectid",
+        "authorsname",
+        "modifieddate",
+        "user_project_info.t_latest_home_visit AS lastvisitdate",
+    ];
+
+    $ordering_criteria = "checkedoutby, modifieddate";
+
+    $result = _run_pp_threshold_query_result($columns, $PPer, $ordering_criteria);
+
+    $projects_grouped_by_PPer = [];
+
+    while($row = mysqli_fetch_assoc($result))
+    {
+        $row_PPer = $row["checkedoutby"];
+        if(!isset($projects_grouped_by_PPer[$row_PPer]))
+        {
+            $projects_grouped_by_PPer[$row_PPer] = [];
+        }
+        $projects_grouped_by_PPer[$row_PPer][] = $row;
+    }
+
+    if($PPer)
+        return array_get($projects_grouped_by_PPer, $PPer, []);
+    else
+        return $projects_grouped_by_PPer;
+}
+
+// internal function to avoid duplicating the SQL logic that determines
+// projects past the PP alert threshold
+function _run_pp_threshold_query_result($columns, $PPer, $ordering_criteria=NULL)
+{
+    global $pp_alert_threshold_days;
+
+    $column_selector = implode(", ", $columns);
+    $user_selector = $PPer ? "AND checkedoutby = '$PPer'" : "";
+    $ordering_criteria_clause = $ordering_criteria ? "ORDER BY $ordering_criteria" : "";
```

Comment:
Sorry if I wasn't clear, what I meant was the renamings:
* $order_by -> $ordering_criteria  [as you've done]
* $order_by_sql -> $order_by_clause  [not $ordering_criteria_clause]

Reply:
That's all.

Reply:
You weren't unclear, I just totally fat-fingered the search/replace.